.github/workflows/test-and-deploy.yml

```diff
@@ -0,0 +1,35 @@
+name: Test and Deploy
+on: push
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: setup node version
+        uses: actions/setup-node@v4
+        with: 
+          node-version-file: .nvmrc
+      - name: npm install 
+        run: npm ci
+      - name: Run tests
+        run: npm test
+  deploy-qa:
+    runs-on: ubuntu-latest
+    permissions: 
+      id-token: write
+      contents: read
+    needs: test
+    if: github.ref == 'refs/head/qa'
```

Comment:
Will the production step come later once this works?

Reply:
yea! just dont want this to functionally deploy but be missing config or something like that.